dashboard/modules/job/job_manager.py

```diff
@@ -594,11 +628,12 @@ async def submit_job(
         # returns immediately and we can catch errors with the actor starting
         # up.
         try:
+            scheduling_strategy = await self._get_scheduling_strategy()
```

Comment:
this always returns the same value btw (unless the key is not found), consider cache this in instance variable if not none?

Reply:
What if the head node goes down and comes up again with a different id (do we need to worry about this case?)